detox/src/devices/drivers/EmulatorDriver.js

```diff
@@ -117,6 +102,55 @@ class EmulatorDriver extends AndroidDriver {
     await telnet.kill();
     await this.emitter.emit('shutdownDevice', { deviceId });
   }
+
+  async _fixEmulatorConfigIniSkinNameIfNeeded(avdName) {
+    const configFile = `${os.homedir()}/.android/avd/${avdName}.avd/config.ini`;
+    const config = ini.parse(fs.readFileSync(configFile, 'utf-8'));
+
+    if (!config['skin.name']) {
+      const width = config['hw.lcd.width'];
+      const height = config['hw.lcd.height'];
+
+      if (width === undefined || height === undefined) {
+        throw new Error(`Emulator with name ${avdName} has a corrupt config.ini file (${configFile}), try fixing it by recreating an emulator.`);
+      }
+
+      config['skin.name'] = `${width}x${height}`;
+      fs.writeFileSync(configFile, ini.stringify(config));
+    }
+    return config;
+  }
+
+  async _getDeviceIdsByType(name) {
+    const device = await this.adb.findDevice(async (candidate) => {
+      const isEmulator = candidate.type === 'emulator';
+      const isMatchingName = candidate.name === name;
+      if (!(isEmulator && isMatchingName)) {
+        return false;
+      }
+
+      // Note: though not entirely pure, this is an important optimization so as to avoid preparsing of all
+      // potentially fit emulators, which can be time consuming - mostly because of the telnet.
+      const isBusy = await this.deviceRegistry._getBusyDevices().includes(candidate.adbName);
```

Comment:
Why use a private method of deviceRegistry? Why not extend and change something else?